### ISSUE TITLE
implement an UNIX socket to send commands

### ISSUE THREAD

**Issue by bmorel** [DaemonEngine/Daemon#700]
Currently, daemonded can only take commands from stdin. I think it would be very helpful to allow it to read commands from an AF_UNIX socket, so that it could be run under a supervisor and still be driven.

**Comment by slipher** [comment]
It already does that. You can use it by launching Daemon with plus commands in the command line while an existing instance is already running. Though it may be broken on Linux currently.

**Comment by illwieckz** [comment]
It works, we fixed the last remaining bugs as far as I know.

**Comment by illwieckz** [comment]
What we may want is a command line option to not start the engine if there is no already running engine, to make sure it is only used as a command forwarder.

**Comment by bmorel** [comment]
Well, then, let me rephrase it: I would like to have a named pipe, that can be changed from command line. This would allow to run the daemon under supervision while still be able to send it commands.
If this exists, then it needs documentation (lol)

**Comment by illwieckz** [comment]
There is no named pipe to send command with cat, but there is a pipe the engine uses to send command to existing instance.

```sh
# This starts an engine in background
nohup ./daemonded +devmap plat23 &

# This forward a command to the already running engine
./daemonded +devmap parpax
```

**Comment by bmorel** [comment]
That's useless.
What server owners need is a way to drive the daemonded daemon without a process constantly running. Running under `screen` is just a hack, a workaround. This ticket is about implementing a proper solution.

**Comment by bmorel** [comment]
Engine already is bad with that "by build default" crash_handler, it's very UNfriendly to server admins that automate things imo. I think it would be nice to improve that, even if opt-in.
I'm ok to implement it myself, if it's accepted as a wished feature, but not important enough to merit senior's attention.

**Comment by necessarily-equal** [comment]
Sounds useful.

**Comment by DolceTriade** [comment]
I don't understand? You always need an instance of daemonded running if you want to interact with it. The 2nd instance doesn't continue to run. It runs the command and exists immediately.

**Comment by bmorel** [comment]
On my server, I used to have 4 instances of daemonded: SK3, SK5, SK7, and SK9. How can I send commands to a specific instance with the current way?

**Comment by necessarily-equal** [comment]
> How can I send commands to a specific instance with the current way?

by setting `-homepath` I bet

**Comment by bmorel** [comment]
this is certainly very practical.

**Comment by DolceTriade** [comment]
Correct. The homepath contains the socket, and daemon will look there.

**Comment by slipher** [comment]
Closing since the feature already exists.
